### PR TITLE
Split overriding into multiple methods

### PR DESCRIPTION
**Pull request by artsiomtserashkovich**

### CODE REVIEW

File: src/neptune_experimental/operation_error_processor.py [hunk 1]
```diff
@@ -29,11 +29,11 @@
 from neptune.internal.utils.logger import logger
 
 from neptune_experimental.env import NEPTUNE_SAMPLE_SERIES_STEPS_ERRORS
-from neptune_experimental.utils import override
+from neptune_experimental.utils import override_attr
 
 
 def initialize() -> None:
-    override(obj=AsyncOperationProcessor, attr="ConsumerThread", target=CustomConsumerThread)
+    override_attr(obj=AsyncOperationProcessor, attr="ConsumerThread", new_attr=CustomConsumerThread)
```
Comment (artsiomtserashkovich) [inline src/neptune_experimental/operation_error_processor.py:36]: As I understand it will be revoked, so ``override_attr`` will be removed in future

Comment (Raalsky) [reply src/neptune_experimental/operation_error_processor.py:36]: Yes

File: src/neptune_experimental/utils.py [hunk 2]
```diff
@@ -22,11 +22,25 @@
 )
 
 
-def override(*, obj: Any, attr: str, target: Callable[..., Any]) -> None:
-    source = getattr(obj, attr)
+def wrap_method(*, obj: Any, method: str, wrapper: Callable[..., Any]) -> None:
+    def wrapped_method_factory(func: Callable[..., Any]) -> Callable[..., Any]:
```
Comment (Raalsky) [inline src/neptune_experimental/utils.py:26]: Looks like it's possible to simplify it with some lambda, don't you think?

**Review by Raalsky — APPROVED** [review]

File: src/neptune_experimental/utils.py (hunk 2 above)
Comment (artsiomtserashkovich) [reply src/neptune_experimental/utils.py:26]: Looks like it's not possible due to ``wraps`` annotation

**Pull request merged**